### ISSUE TITLE
 `import ecoscope` takes too long 

### ISSUE THREAD

**Issue by cisaacstern** [wildlife-dynamics/ecoscope#142]
Currently importing ecoscope takes ~6 seconds.

```console
$ time python3 -c "import ecoscope"
python3 -c "import ecoscope"  6.07s user 0.94s system 80% cpu 8.673 total
```

If I comment out this line

https://github.com/wildlife-dynamics/ecoscope/blob/b3b9d1543eee5357ef9b2df8a4135d47d40f39d4/ecoscope/__init__.py#L1

that is reduced to 0.3 seconds:

```console
$ time python3 -c "import ecoscope"
python3 -c "import ecoscope"  0.03s user 0.01s system 73% cpu 0.060 total
```

IMHO 6 seconds is too long to wait for ecoscope to import. An example of why this matters is in test development: if test session takes a minimum of 6 seconds to initialize, that adds up to a lot of developer time over the course of hundreds or thousands of test invocations.

I am going to see if I can reduce import time and open a PR for review 😃

**Comment by cisaacstern** [comment]
A little more detail on import time of the various components:

- `from ecoscope import analyis` ➡️  4s
- `from ecoscope import base`  ➡️  circular import w analysis, can't be imported on it's own (side note: this seems like its own issue, i would think things named `base` don't have other dependencies within the package)
- `from ecoscope import contrib`  ➡️ 0.2s
- `from ecoscope import io`  ➡️ 0.98s
- `from ecoscope import mapping` ➡️ 0.58s
- `from ecoscope import plotting` ➡️ 1.92s

4 + 0.2 + 0.98 + 0.58 + 1.92 = 7.68 seconds (so these numbers check out)

**Comment by cisaacstern** [comment]
> I am going to see if I can reduce import time and open a PR for review 😃

Note, for now to unblock development of #141, I've just commented out this line:

https://github.com/wildlife-dynamics/ecoscope/blob/818696e5ad69e216c902ac73e180bcd0753588b2/ecoscope/__init__.py#L1-L2

I can revisit a more general fix soon!

**Comment by atmorling** [comment]
I've [tested](https://github.com/atmorling/ecoscope/tree/142/slow_import) the following:

- Moving the functions `analysis/astronomy.py` into `base/utils.py` to solve the circular import issue
- Deferring expensive imports like `mapclassify` and `astroplan`
- Rearranging the imports in `ecoscope/__init__.py` so that `analysis` is loaded last

This reduces the import time to around ~2.7 seconds. 
![screenshot](https://github.com/wildlife-dynamics/ecoscope/assets/24975882/86261765-6251-4753-9dc5-f5e516b035b4)

Deferring the load of `ecoscope.analysis.UD` and `ecoscope.mapping` (removing their imports from the relevant `__init__.py`s) gets this down to ~1.3 seconds. This means one would have to explicitly import these modules in addition to `import ecoscope` . 

Deferring the expensive imports isn't exactly PEP8 Approved<sup>TM</sup>, so I think playing around with [lazy imports](https://docs.python.org/3/library/importlib.html#implementing-lazy-imports) is worthwhile.

Also learned that [tuna](https://github.com/nschloe/tuna) for visualizing import profiler outputs. (See the pretty screenshot)

**Comment by cisaacstern** [comment]
amazing progress!
